### PR TITLE
Fix hipcc + lightning_kokkos issue

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [X] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [X] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
This PR fixes an issue with the hipcc (llvm) compiler, which does not allow `[[maybe_unused]]` attributes in kokkos lambdas, and fails to correctly resolve `pennylane::utils::exp2` function. E.g. in nightly rocm docker build https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/12173013064

**Description of the Change:**
In lightning kokkos, `exp2` are explicitly replaced by `pennylane::utils::exp2`, and `[[maybe_unused]]` attributes are now moved from the lambda arguments to the lambda body.

This has been tested to work with the rocm docker build. (https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/12188525751/job/34001703721)

**Benefits:**
Fixes rocm build failures
**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-79824]

### CODE REVIEW

**Review by LuisAlfredoNu — APPROVED** [review]
Thank @josephleekl for solving this issue.

**Review by multiphaseCFD — APPROVED** [review]
Thanks @josephleekl ! LGTM!

File: pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ template <class Precision> struct multiQubitOpFunctor {
                         const KokkosComplexVector &matrix_,
                         const std::vector<std::size_t> &wires_) {
         wires = vector2view(wires_);
-        dim = exp2(wires_.size());
+        dim = Pennylane::Util::exp2(wires_.size());
```
Comment (maliasadi) [inline pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp:63]: We could also bring this the scope at the top of the file similar to [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/9eb61230724131d3fca036a09d1e84771aafacfa/pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubitManaged.hpp#L39) for a slightly better code quality. What do you think?

**Review by maliasadi — APPROVED** [review]
LGTM :raised_hands:

File: pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp (hunk 1 above)
Comment (josephleekl) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp:63]: that will make it a lot cleaner. I'll update it, thanks!

Comment (josephleekl) [reply pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp]: i've now tried to change it, but hipcc really struggles to resolve the correct exp2 even when putting `using Pennylane::Util::exp2;` at the start... so I will put the namespaces back

**Pull request merged**